base/set.jl

```diff
@@ -103,6 +116,30 @@ const ⊆ = issubset
 ⊊(l::Set, r::Set) = <(l, r)
 ⊈(l::Set, r::Set) = !⊆(l, r)
 
+doc"""
+    unique(C,k::Integer)
+
+Returns up to k distinct values in C in the order they appear
+"""
+function unique(C,k::Integer)
+    out = Vector{eltype(C)}()
+    seen = Set{eltype(C)}()
+    total = 0
+    for x in C
+        if !in(x, seen)
+            push!(seen, x)
+            push!(out, x)
+            total += 1
+            if total == k
+                return out
+            end
+        end
+    end
+    return out
+end
+
+unique(x::Set) = collect(x)
```

Comment:
Will do. There are a couple places where I can;t swap out `Set` in favor of `AbstractSet`.